doc/api/http.md

```diff
@@ -917,6 +917,48 @@ Example:
 var contentType = response.getHeader('content-type');
 ```
 
+### response.getHeaderNames()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Array}
+
+Returns an array containing the names of the current outgoing headers. All
+header names are lowercase.
+
+Example:
+
+```js
+response.setHeader('Foo', 'bar');
+response.setHeader('Set-Cookie', ['foo=bar', 'bar=baz']);
+
+var headerNames = response.getHeaderNames();
+// headerNames === ['foo', 'set-cookie']
+```
+
+### response.getHeaders()
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Object}
+
+Returns a shallow copy of the current outgoing headers. The keys of the
+returned object are the header names and the values are the respective header
+values. Any header values that are arrays are also shallowly cloned. All
+header names are lowercase.
+
+Example:
+
+```js
+response.setHeader('Foo', 'bar');
+response.setHeader('Set-Cookie', ['foo=bar', 'bar=baz']);
+
+var headers = response.getHeaders();
+// headers === { foo: 'bar', 'set-cookie': ['foo=bar', 'bar=baz'] }
+```
```

Comment:
Out of curiosity: did you investigate the possibility of having a variation on this that returns an iterator instead? Not sure if that would perform better or worse than a copy if we allowed it to do a live iteration. Obviously it depends entirely on how we expect users to make use of this.

Also, a `response.hasHeader(name)` that returns `boolean` might be worthwhile for the cases someone wishes to simply check that a header exists.